echo/test/unit/echo_client_test.js

```diff
@@ -399,23 +304,18 @@ describe("Echo Node Client Test Suite", function(){
                 class: 'testclass'
             };
 
-            var requestStub = sandbox.stub(request, 'get')
-            requestStub.callsFake(function (options, callback) {
-                options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                callback(
-                  null,
-                  {statusCode: 401},
-                  {
-                      "error": "invalid_token"
-                  }
-              );
-            });
+            nock(endPoint)
+                .get('/1/analytics/sum?class=testclass')
+                .reply(function(uri, requestBody){
+                    this.req.headers['user-agent'].should.equal('talis-node/0.2.3');
+                    return [401, {"error": "invalid_token"}];
+                });
 
             echoClient.queryAnalytics("incorrect_token", "sum", params, false, function(err, result) {
                 (err === null).should.be.false;
 
-                err.code.should.equal(8);
-                err.label.should.equal("INVALID_TOKEN");
+                err.code.should.equal(6);
+                err.label.should.equal("UNAUTHORISED");
```

Comment:
8 / INVALID_TOKEN -> 6 / UNAUTHORIZED

The mocking is returning the same 401 response. Is this something caused by the change of http client library?

This isn't talking about AUTHENTICATION/AUTHORISATION - so what is the difference between INVALID_TOKEN and UNAUTHORISED?